webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -975,6 +1098,83 @@ public static ByteArrayOutputStream requestsToZip(ServletContext context, String
 
         ZipOutputStream zipOut = new ZipOutputStream(baosZip);
         HashMap<String, Boolean> seenTitles = new HashMap<String, Boolean>();
+
+        boolean fmcBeingHeld = false;
+        for(ScrambleRequest scrambleRequest : scrambleRequests) {
+            if(scrambleRequest.fmc) {
+                fmcBeingHeld = true;
+
+                String safeTitle = toFileSafeString(scrambleRequest.title);
+                int salt = 0;
+                String tempNewSafeTitle = safeTitle;
+                while(seenTitles.get(tempNewSafeTitle) != null) {
+                    tempNewSafeTitle = safeTitle + " (" + (++salt) + ")";
+                }
+                safeTitle = tempNewSafeTitle;
+                seenTitles.put(safeTitle, true);
```

Comment:
This feels weird to me, because this isn't actually the title of the file you put in the zip file (later on, you append " Scramble Cutout Sheet.pdf"). Won't this cause problems later on when generating the non-cutout sheet?

edit: Confirmed by generated a zip at this url: http://localhost:2014/scramble/#competitionName=Scrambles%20for%202017-04-25&rounds=i('eventID'-'333fm'_'round'-'1'_'groupCount'-2_'scrambleCount'-3_'extraScrambleCount'-0_'copies'-1)_('eventID'-'333fm'_'round'-'2'_'groupCount'-2_'scrambleCount'-3_'extraScrambleCount'-0_'copies'-1)!&version=1.0:

```
~/Downloads @kaladin> unzip -l Scrambles\ for\ 2017-04-25.zip 
Archive:  Scrambles for 2017-04-25.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     2771  2017-04-25 10:53   pdf/3x3x3 Fewest Moves Round 1 Group A Scramble Cutout Sheet.pdf
     2206  2017-04-25 10:53   pdf/3x3x3 Fewest Moves Solution Sheet.pdf
     9746  2017-04-25 10:53   pdf/3x3x3 Fewest Moves Round 1 Group A (1).pdf
      211  2017-04-25 10:53   txt/3x3x3 Fewest Moves Round 1 Group A (1).txt
     9734  2017-04-25 10:53   pdf/3x3x3 Fewest Moves Round 1 Group B.pdf
      214  2017-04-25 10:53   txt/3x3x3 Fewest Moves Round 1 Group B.txt
     9740  2017-04-25 10:53   pdf/3x3x3 Fewest Moves Round 2 Group A.pdf
      210  2017-04-25 10:54   txt/3x3x3 Fewest Moves Round 2 Group A.txt
     9725  2017-04-25 10:54   pdf/3x3x3 Fewest Moves Round 2 Group B.pdf
      216  2017-04-25 10:54   txt/3x3x3 Fewest Moves Round 2 Group B.txt
     2007  2017-04-25 10:54   Scrambles for 2017-04-25.json
     2029  2017-04-25 10:54   Scrambles for 2017-04-25.jsonp
      550  2017-04-25 10:54   Scrambles for 2017-04-25.html
    37943  2017-04-25 10:54   Scrambles for 2017-04-25.pdf
---------                     -------
    87302                     14 files
```

Note how the "... Round 1 Group A (1).pdf" has that unecessary (1) in it.

Also note that Round 1 Group B is missing, as is Round 2 Group A and Round 2 Group B.